dask/dataframe/core.py

```diff
@@ -3349,6 +3353,7 @@ def map_partitions(func, *args, **kwargs):
     name = '{0}-{1}'.format(name, token)
 
     from .multi import _maybe_align_partitions
+    args, args_dasks = _process_lazy_args(args)
```

Comment:
I *think* this can be moved down to where the call to `to_task_dask` is with no issues (since alignment also ignores `Delayed` objects). I'd prefer to group those together if possible.